libretro/libretro-core.c

```diff
@@ -1417,34 +1454,187 @@ void retro_set_environment(retro_environment_t cb)
          {
             { "none", "disabled" },
-            { "small", "Small" },
-            { "medium", "Medium" },
-            { "maximum", "Maximum" },
+            { "small", "Small Crop" },
+            { "medium", "Medium Crop" },
+            { "full", "Full Crop" },
+  #if defined(__X64__) || defined(__X64SC__) || defined(__X128__)   // only include manual cropping for c64/128
+           { "manual", "Manual Crop" },
+  #endif
             { NULL, NULL },
          },
          "none"
       },
+
+  #if defined(__X64__) || defined(__X64SC__) || defined(__X128__)
+      {
+         "vice_bordercrop_top",         // Manual Pixel settings for Top Cropping option
+         "Manual Top Cropping",
+         "",
+         {
+            { "0", "0 px" },
+            { "1", "1 px" },
+            { "2", "2 px" },
+            { "3", "3 px" },
+            { "4", "4 px" },
+            { "5", "5 px" },
+            { "6", "6 px" },
+            { "7", "7 px" },
+            { "8", "8 px" },
+            { "9", "9 px" },
+            { "10", "10 px" },
+            { "11", "11 px" },
+            { "12", "12 px" },
+            { "13", "13 px" },
+            { "14", "14 px" },
+            { "15", "15 px" },
+            { "16", "16 px" },
+            { "17", "17 px" },
+            { "18", "18 px" },
+            { "19", "19 px" },
+            { "20", "20 px" },
+            { "21", "21 px" },
+            { "22", "22 px" },
+            { "23", "(NTSC Border) 23 px" },
+            { "24", "24 px" },
+            { "25", "25 px" },
+            { "26", "26 px" },
+            { "27", "27 px" },
+            { "28", "28 px" },
+            { "29", "29 px" },
+            { "30", "30 px" },
+            { "31", "31 px" },
+            { "32", "32 px" },
+            { "33", "33 px" },
+            { "34", "34 px" },
+            { "35", "(PAL Border) 35 px" },
+            { "36", "36 px" },
+            { "37", "37 px" },
+            { "38", "38 px" },
+            { "39", "39 px" },
+            { "40", "40 px" },
+            { "41", "41 px" },
+            { "42", "42 px" },
+            { "43", "43 px" },
+            { "44", "44 px" },
+            { "45", "45 px" },
+            { "46", "46 px" },
+            { "47", "47 px" },
+            { "48", "48 px" },
+            { "49", "49 px" },
+            { "50", "50 px" },
+            { "51", "51 px" },
+            { "52", "52 px" },
+            { "53", "53 px" },
+            { "54", "54 px" },
+            { "55", "55 px" },
+            { "56", "56 px" },
+            { "57", "57 px" },
+            { "58", "58 px" },
+            { "59", "59 px" },
+            { "60", "60 px" },      // same as variable max_crop_pixels
+            { NULL, NULL },
+         },
+         "none"
+      },
+
+      {
+         "vice_bordercrop_bottom",          // Manual Pixel settings for Bottom Cropping option
+         "Manual Bottom Cropping",

```

Comment:
This is too complicated to use IMO. Just provide a few default sensible crop options. Not even standalone Vice has this granularity. Why would it be necessary?

Reply:
Why would you target the inabilities of stand-alone vice?